src/TorchSharp/Tensor/Tensor.Factories.cs

```diff
@@ -1369,10 +1369,18 @@ public static Tensor tensor(IList<bool> rawArray, long[] dimensions, torch.Scala
         /// <summary>
         /// Create a tensor from an array of values, shaping it based on the shape passed in.
         /// </summary>
-        public static Tensor tensor(bool[] dataArray, long[] dimensions, torch.ScalarType? dtype = null, torch.Device? device = null, bool requiresGrad = false)
+        public static Tensor tensor(bool[] rawArray, long[]? dimensions = null, torch.ScalarType? dtype = null, torch.Device? device = null, bool requiresGrad = false)
+        {
+            if (dimensions is null)
+                dimensions = new long[] { rawArray.LongLength };
```

Comment:
>new long[]  [](http://example.com/codeflow?start=29&length=11)

can you allocate this on the stack instead?

Reply:
Yeah, I was just thinking about that. It should work.

Reply:
So, I would have to use a fixed block, to pass the shape, right? It should probably do that, anyway, just not correctly implemented today.

Reply:
I think `Span<long> dimensions = stackalloc long[1]` would work just fine. 👍

Reply:
If you area going to use pointers, like allocation using stackalloc, then you don't need fixed block but you will use unsafe block. 

If you use Span, you will not need fixed nor unsafe block.

Reply:
> If you area going to use pointers, like allocation using stackalloc, then you don't need fixed block but you will use unsafe block.
> 
> If you use Span, you will not need fixed nor unsafe block.

I think `Span<>` should be fine here. It is practically the same thing, and you avoid the dirty `unsafe` keyword.

Reply:
I've only ever used pointer when passing arrays to native code -- can I go straight from a ReadOnlySpan to IntPtr without unsafe or fixed?

Reply:
`(IntPtr)MemoryMarshal.GetReference(buffer)` where buffer is the span.